packages/experimental/TabList/src/TabListAnimatedIndicator/useAnimatedIndicatorStyles.ts

```diff
@@ -1,27 +1,68 @@
   const { animatedIndicatorStyles, layout, selectedKey, setCanShowAnimatedIndicator, vertical } = React.useContext(TabListContext);
 
-  const selectedIndicatorLayout = React.useMemo<TabLayoutInfo | null>(() => {
-    return selectedKey ? layout.tabs[selectedKey] : null;
-  }, [selectedKey, layout.tabs]);
+  // animated values
+  const indicatorTranslate = React.useRef(new Animated.Value(0)).current;
+  const indicatorScale = React.useRef(new Animated.Value(1)).current;
+
+  const [startingIndicatorLayout, setStartingIndicatorLayout] = React.useState<TabLayoutInfo | null>(null);
+
+  React.useEffect(() => {
+    if (startingIndicatorLayout === null && layout.tabs[selectedKey]) {
+      setStartingIndicatorLayout(layout.tabs[selectedKey]);
+    }
+  }, [selectedKey, layout.tabs, startingIndicatorLayout, setStartingIndicatorLayout]);
+
+  React.useEffect(() => {
+    const selectedIndicatorLayout = layout.tabs[selectedKey];
+    if (startingIndicatorLayout && selectedIndicatorLayout) {
+      /**
+       * Calculate transforms. Because the scale transform's origin is at the center, we need to calculate an extra offset to add to the
+       * translate transform to place the indicator at the correct location on screen.
+       */
+      let scaleValue: number, translateValue: number, translateOffset: number;
+      if (vertical) {
+        scaleValue = selectedIndicatorLayout.height / startingIndicatorLayout.height;
+        translateValue = selectedIndicatorLayout.y - startingIndicatorLayout.y;
+        translateOffset = (selectedIndicatorLayout.height - startingIndicatorLayout.height) / 2;
+      } else {
+        scaleValue = selectedIndicatorLayout.width / startingIndicatorLayout.width;
+        translateValue = selectedIndicatorLayout.x - startingIndicatorLayout.x;
+        translateOffset = (selectedIndicatorLayout.width - startingIndicatorLayout.width) / 2;
+      }
+      Animated.timing(indicatorScale, {
+        toValue: scaleValue,
+        duration: 200,
+        useNativeDriver: false,
+      }).start();
+      Animated.timing(indicatorTranslate, {
+        toValue: translateValue + translateOffset,
+        duration: 200,
+        useNativeDriver: false,
+      }).start();
+    }
+  }, [indicatorScale, indicatorTranslate, layout.tabs, selectedKey, startingIndicatorLayout, vertical]);
 
   // Calculate styles using both layout information and user defined styles
   const styles = React.useMemo<AnimatedIndicatorStyles | null>(() => {
     // if not all layout props have been recorded for the current selected indicator, don't render the animated indicator
-    if (!selectedIndicatorLayout) {
+    if (!startingIndicatorLayout) {
       return null;
     }
-    const { x, y, width, height, startMargin, tabBorderWidth } = selectedIndicatorLayout;
+    const { x, y, width, height, startMargin, tabBorderWidth } = startingIndicatorLayout;
     const containerStyles: ViewStyle = {
       position: 'absolute',
```

Comment:
ahhhh because the container here only hosts the animated indicator and not the static indicator (part of the tab I think?)?

FWIW `z-index` is only needed for re-ordering purposes; on all platforms if you have two elements overlapping without any use of `z-index` the second element would be drawn over the first, third over both, etc.

Reply:
> ahhhh because the container here only hosts the animated indicator and not the static indicator (part of the tab I think?)?

That's correct. is the use of absolute positioning still an issue?

Reply:
not a problem just clarifying understanding of the structure :)